### PR TITLE
[SearchBox] Fix flickering due to debounce

### LINKED ISSUE #408: Searchbox flickering/ Cypress inestability
https://github.com/cypress-io/cypress/issues/5480

### PR DESCRIPTION
**Pull request by SferaDev**
* **Issue:** Closes https://github.com/EyeSeeTea/metadata-synchronization/issues/408

- Changed to _.debounce (we can remove the ``throttle-debounce`` dependency as soon as we deprecate the old objects table).

### CODE REVIEW

File: src/data-table/ObjectsTable.tsx [hunk 1]
```diff
@@ -78,35 +78,55 @@ export function ObjectsTable<T extends ReferenceObject = TableObject>(props: Obj
...
-                  <SearchBox
-                      value={searchValue}
-                      hintText={searchBoxLabel || "Search items"}
-                      onChange={handleSearchChange}
-                  />
-              </div>,
-              parentFilterComponents,
-          ]
-        : parentFilterComponents;
+    const filterComponents = useMemo(
```
Comment (tokland) [inline src/data-table/ObjectsTable.tsx:89]: That's the way I am also creating in-line memoized components, seems to work well.

**Review by tokland — APPROVED** [review]
Code ok, tested with project-monitoring-app, also ok. :+1: 

I've seen some problems with i18n, but that's for another PR:

https://github.com/EyeSeeTea/d2-ui-components/issues/141

**Comment by xurxodev** [comment]
The blink seems to be fixed. However the tests continue to fail, so the cypress bug is confirmed.
cypress-bug

**Comment by SferaDev** [comment]
> The blink seems to be fixed. However the tests continue to fail, so the cypress bug is confirmed.
> cypress-bug

@xurxodev Great, thanks for confirming!

We will need to track that cypress issue then...

**Pull request merged**